From 988278f01a17da4d513ca477e2c2e001c34722fd Mon Sep 17 00:00:00 2001
From: Matt Mitchell <mmitche@microsoft.com>
Date: Tue, 10 Sep 2019 10:47:51 -0700
Subject: [PATCH 10/11] Compute IsShipping* before version strings are computed
 (#3916)

The IsShipping properties were previously computed in ProjectDefaults.targets, which is imported after Version.BeforeCommonTargets.targets.  This means you can't use IsShippingPackage to determine how to version. Move this computation to Version.BeforeCommonTargets.targets.
---
 .../tools/ProjectDefaults.targets             | 17 ---------------
 .../tools/Version.BeforeCommonTargets.targets | 21 +++++++++++++++++++
 2 files changed, 21 insertions(+), 17 deletions(-)

diff --git a/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.targets b/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.targets
index 6e519727..138d5954 100644
--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.targets
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.targets
@@ -12,23 +12,6 @@
   <Target Name="Deploy" AfterTargets="Build" Condition="'$(DeployProjectOutput)' == 'true'" />
 
   <PropertyGroup>
-    <!--
-      Unless specified otherwise project is assumed to produce artifacts (assembly, package, vsix, etc.) that ship.
-      Test projects automatically set IsShipping to false.
-
-      Some projects may produce packages that contain shipping assemblies but the packages themselves do not ship.
-      Thes projects shall specify IsShippingPackage=false and leave IsShipping unset (will default to true).
-
-      Targets that need to determine whether an artifact is shipping shall use the artifact specific IsShippingXxx property,
-      if available for the kind of artifact they operate on.
-    -->
-    <IsShipping Condition="'$(IsShipping)' == ''">true</IsShipping>
-
-    <IsShippingAssembly Condition="'$(IsShippingAssembly)' == ''">$(IsShipping)</IsShippingAssembly>
-    <IsShippingPackage Condition="'$(IsVisualStudioBuildPackage)' == 'true'">false</IsShippingPackage>
-    <IsShippingPackage Condition="'$(IsShippingPackage)' == ''">$(IsShipping)</IsShippingPackage>
-    <IsShippingVsix Condition="'$(IsShippingVsix)' == ''">$(IsShipping)</IsShippingVsix>
-
     <!--
       Set PackageOutputPath based on the IsShippingPackage flag set by projects.
       This distinction allows publishing tools to determine which assets to publish to official channels.
diff --git a/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets b/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets
index 0808c222..f3340a5a 100644
--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets
@@ -2,6 +2,27 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project>
 
+  <!--
+    Compute the IsShipping* properties
+
+    Unless specified otherwise project is assumed to produce artifacts (assembly, package, vsix, etc.) that ship.
+      Test projects automatically set IsShipping to false.
+
+      Some projects may produce packages that contain shipping assemblies but the packages themselves do not ship.
+      Thes projects shall specify IsShippingPackage=false and leave IsShipping unset (will default to true).
+
+      Targets that need to determine whether an artifact is shipping shall use the artifact specific IsShippingXxx property,
+      if available for the kind of artifact they operate on.
+  -->
+  <PropertyGroup>
+    <IsShipping Condition="'$(IsShipping)' == ''">true</IsShipping>
+
+    <IsShippingAssembly Condition="'$(IsShippingAssembly)' == ''">$(IsShipping)</IsShippingAssembly>
+    <IsShippingPackage Condition="'$(IsVisualStudioBuildPackage)' == 'true'">false</IsShippingPackage>
+    <IsShippingPackage Condition="'$(IsShippingPackage)' == ''">$(IsShipping)</IsShippingPackage>
+    <IsShippingVsix Condition="'$(IsShippingVsix)' == ''">$(IsShipping)</IsShippingVsix>
+  </PropertyGroup>
+
   <!--
     Specification: https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/Versioning.md
 
-- 
2.18.0

